scripts/products.js

```diff
@@ -0,0 +1,93 @@
+import {
+  getMetadata,
+} from './lib-franklin.js';
+
+function querySelectorIncludesText(element, selector, text) {
+  return Array.from(element.querySelectorAll(selector))
+    .find((el) => el.textContent.includes(text));
+}
+
+function applyBackgroundImage(elementContainingSectionMetadata, elementToApplyBG) {
+  const sectionMetadata = elementContainingSectionMetadata.querySelector(('.section-metadata'));
+  if (sectionMetadata) {
+    sectionMetadata.remove();
+    const backgroundDiv = querySelectorIncludesText(sectionMetadata, 'div>div>div', 'background');
+    const styleDiv = querySelectorIncludesText(sectionMetadata, 'div>div>div', 'style');
+    const style = styleDiv.nextElementSibling.textContent;
+
+    let url = backgroundDiv.nextElementSibling.querySelector('img').src;
+    url = url.substring(0, url.indexOf('?'));
+    url += '?width=2000&format=webply&optimize=medium';
+
+    elementToApplyBG.classList.add('product-background');
+    elementToApplyBG.style.backgroundPosition = style;
+    elementToApplyBG.style.backgroundImage = `url(${url})`;
+  }
+}
+
+/**
+ * Loads the product template and replaces the content when the metadata is set accordingly
+ * @param {Element} doc The container element
+ * @returns {Promise}
+ */
+// eslint-disable-next-line import/prefer-default-export
+export async function loadProduct(doc) {
+  const templatePath = getMetadata('template');
```

Comment:
I think its better to do the template check in scripts.js and only if it is the right template to load products.js - otherwise we have to load it on every page, regardless if it is used or not